packages/ui/src/bounty/modals/AnnounceWorkEntryModal/AnnounceWorkEntryModal.tsx

```diff
@@ -36,14 +38,21 @@ import { WaitModal } from '@/common/components/WaitModal'
 import { Fonts } from '@/common/constants'
 import { useApi } from '@/common/hooks/useApi'
 import { useModal } from '@/common/hooks/useModal'
+import { useSchema } from '@/common/hooks/useSchema'
 import { formatTokenValue } from '@/common/model/formatters'
 import { MemberInfo } from '@/memberships/components'
 import { useMyMemberships } from '@/memberships/hooks/useMyMemberships'
 import { BindStakingAccountModal } from '@/memberships/modals/BindStakingAccountModal/BindStakingAccountModal'
 import { SwitchMemberModalCall } from '@/memberships/modals/SwitchMemberModal'
+import { IStakingAccountSchema, StakingAccountSchema } from '@/memberships/model/validation'
 
 const transactionSteps = [{ title: 'Bind staking account' }, { title: 'Announce Work' }]
 
+const schema = Yup.object().shape({
+  amount: Yup.string(),
+  account: StakingAccountSchema,
```

Comment:
Because the goal is to remove as much duplicated logic as possible, if possible I think the `account` should be required (does `StakingAccountSchema.required` works ?) and at the end:
```ts
<ButtonPrimary size="medium" disabled={!isValid} onClick={nextStep}>
```
With `isValid` being the value returned by `useSchema`. This way all the logic linked to `isValidNext` can be removed and YUP takes care of all the validation.

WDYT ?

Reply:
Only question is whether we want this issue to just solve the bugs or addinally straight out the logic. I'll be cleaning every bounty modal in separate issue, so I thought I'll leave it out not to overcomplicate PR.

Reply:
Got it, I thought you wanted to split the "cleaning" one modal at a time.
But if this PR is just about fixing #2739, lets leave this suggestion and the `<SelectAccount />` filter for another PR.